src/main/java/com/urbanairship/datacube/Op.java

```diff
@@ -6,16 +6,18 @@
 
 import com.urbanairship.datacube.ops.LongOp;
 
+import java.util.function.Supplier;
+
 
 /**
  * A cell mutation or cell bucket. For example, a cube storing counters would contain Ops that
  * are numbers (see e.g. {@link LongOp}).
  */
-public interface Op extends CSerializable {
+public interface Op<T> extends CSerializable, Supplier<T> {
```

Comment:
this seems like an easy win to me-- you gain guarantees that you are combining compatible ops.